src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java

```diff
@@ -29,35 +31,221 @@ public class GameController {
+      if(!userService.isGameBelongingToUser(game, user)) {
+        throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Wrong use of API call: gameId must belong to userId");
+      }
+
+      // delete play
+      game = gameService.deletePlay(game, otherUserId);
+      // a game can be empty, thus we delete it (if not active)
+      userService.deleteGameIfEmpty(user, game);
+      // Create a match between the users if we like each other
+      if(userService.otherUserLikesUser(user, otherUser) && userLikesOtherUser){
+        Match match = userService.createMatch(user, otherUser);
+        userService.setMatch(match);
+        UserGetDTO userGetDTO = DTOMapper.INSTANCE.convertEntityToUserGetDTO(otherUser);
+        return new ResponseEntity<>(userGetDTO, null, HttpStatus.CREATED);
+      } else {
+        // only set like if we haven't a match - handles multiple identical requests
+        if (!userService.doesMatchExist(user, otherUser)){
+          userService.setUserLikesOtherUser(user, otherUser);
+        }
+        return new ResponseEntity<>(null, null, HttpStatus.NO_CONTENT);
+      }
+    }
+
+    /**
+     * create Play of a player
+     */
+    @PostMapping("/users/{userId}/whiteCards/{cardId}")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseBody
+    public void submitWhiteCard(@RequestHeader(value = "authorization", required = false) String token,
+                            @PathVariable(value = "userId") Long id,
+                            @PathVariable(value = "cardId") Long cardId,
+                            @RequestBody GameIDPostDTO gameIDPostDTO){
+
+      userService.checkSpecificAccess(token, id); // throws 401, 404
+
+      // get the game that is voted on
+      Game inputGame = DTOMapper.INSTANCE.convertGameIDPostDTOToEntity(gameIDPostDTO);
+      Game game = gameService.getGameById(inputGame.getId());
+
+      // make sure that the game does not to the caller himself, you shall not give a white card to your own game.
+      if(userService.isGameBelongingToUser(game, userService.getUserById(id))) {
+        throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                "Wrong use of API call: gameId must belong to userId");
+      }
+      // get the played white card
+      WhiteCard whiteCard = gameService.getWhiteCardById(cardId);
+      // check if the white card is one that has been assigned to user and not a random one
+      if(!userService.isWhiteCardBelongingToUser(whiteCard, id)){
+        throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                "Wrong use of API call: You can only submit your white cards. Call \n" +
+                        "GET /users/" + id + "games/whiteCards \n to get your cards!");
+      }
+      // create a play
+      Play play = gameService.createPlay(id, cardId);
+      // only assign the play to the game if user has not already voted on the game (he won't lose his white card if he has)
```

Comment:
Nope, the scenario is the following: a user plays a white card on a game, and we want to prevent that the user plays another white card (or the same) on that game (He shouldn't receive the same game again, but let's make the backend also safe for other API calls than from client).